sdk/src/wallet/wallet/operations/stronghold_backup/mod.rs

```diff
@@ -148,6 +138,17 @@ impl Wallet {
 
         stronghold_password.zeroize();
 
+        // drop secret manager, otherwise we get a deadlock in set_client_options() (there inside of save_wallet_data())
+        drop(secret_manager);
```

Comment:
What is the reason for the deadlock?

Reply:
Oh, we're holding the write lock above, nvm.

Reply:
In restore_backup() we write() on the secret_manager to be able to replace it with the restored one and in save_wallet_data we have `secret_manager.read().await;` to convert the SecreManager to its Dto